docs/tutorials/spec2006-tutorial-details.md

```diff
@@ -0,0 +1,636 @@
+```
+
+To build the disk image,
+
+```sh
+cd disk-image/
+./packer validate spec-2006/spec-2006.json # validate the script, including checking the input files
+./packer build spec-2006/spec-2006.json
+```
+
+The process should take about than an hour to complete on a fairly recent machine with a cable internet speed.
+The disk image will be in `disk-image/spec-2006/spec-2006-image/spec-2006`.
+
+**Note:** Packer will output a VNC port that could be used to inspect the building process.
+Ubuntu has a built-in VNC viewer, namely Remmina.
+
+**Note:** [More about using packer and building disk images](../main-doc/disks.md).
+
+Now, in launch_spec2006_experiments.py, we make an Artifact object of the disk image.
+
+```python
+disk_image = Artifact.registerArtifact(
+    command = './packer build spec-2006/spec-2006.json',
+    typ = 'disk image',
+    name = 'spec-2006',
+    cwd = 'disk-image/',
+    path = 'disk-image/spec-2006/spec-2006-image/spec-2006',
+    inputs = [packer, experiments_repo, m5_binary,],
+    documentation = 'Ubuntu Server with SPEC 2006 installed, m5 binary installed and root auto login'
+)
+```
+
+### Compiling Linux Kernel
+In this step, we will download Linux kernel source code and compile the Linux kernel.
+The file of interest in this step is the vmlinux file.
+
+First, we download the Linux kernel source code.
+Version 4.19.83 has been tested with gem5 as discussed [in the other tutorial](boot-tutorial.md).
+We suggest using config files that have been tested with gem5.
+The following command will shallow clone the linux stable repository as well as checking out the tag v4.19.83, which contains the code for linux kernel version 4.19.83.
+The git command also works well for other version numbers.
+
+In the root of the experiment folder,
+
+```sh
+git clone --branch v4.19.83 --depth 1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/
+mv linux linux-4.19.83
+```
+
+Now, in launch_spec2006_experiments.py, we make an Artifact object of the Linux stable git repo.
+
+```python
+linux_repo = Artifact.registerArtifact(
+    command = '''
+    	git clone git clone --branch v4.19.83 --depth 1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/;
+    	mv linux linux-4.19.83
+    ''',
+    typ = 'git repo',
+    name = 'linux-4.19.83',
+    path =  'linux-4.19.83',
+    cwd = './',
+    documentation = 'Linux kernel 4.19 source code repo obtained in November'
+)
+```
+
+Next, we compile the Linux kernel.
+We will make a folder named linux-configs containing all working linux configs.
+Working Linux configs and documentations for generating a Linux config are discussed here [here](boot-tutorial.md).
+
+In the root folder of the experiment,
+
+```sh
+mkdir linux-configs
+```
+
+To download the linux-4.19.83 configs,
+
+```sh
```

Comment:
Again, I don't hate it done this way, but I'm wondering that maybe a better step would just be to pull the gem5-resources repo and work from there? It's not a big issue for me, but we seem to be wgetting a lot.

Reply:
ditto